controller/controller.go

```diff
@@ -54,8 +60,18 @@ type Config struct {
 	// list of resources being executed for the received runtime object can be
 	// versioned and different resources can be executed depending on the runtime
 	// object being reconciled.
-	ResourceSets         []*ResourceSet
-	RuntimeObjectFactory func() pkgruntime.Object
+	ResourceSets []*ResourceSet
+	// NewRuntimeObjectFunc returns a new initialized pointer of a type
+	// implementing the runtime object interface. The object returned is used with
+	// the controller-runtime client to fetch the latest version of the object
+	// itself. That way we can manage all runtime objects in a somewhat generic
+	// way. See the example below.
+	//
+	//     func() pkgruntime.Object {
+	//        return new(corev1.ConfigMap)
+	//     }
```

Comment:
Let's please use tabs in comments as well.

Reply:
We never did this and never agreed on it, plus, I hate it. I do not want to mess with any tabs in my editor. At the moment I lack proper configuration in my dev environment for this. Tabs vs spaces always fucks around with me.

Reply:
When this is important to you, get `fmt` sorted and I try to comply.

Reply:
In Go tabs are used for indentation. No personal feelings involved.

Reply:
Yes, and the go toolchain does it automatically where it apparently cares about, in the actual source code. Comments are free form AFAIK. I am all for more consistency and do not stand in your way here.